cmake/toolchains/ARM.cmake

```diff
@@ -13,6 +19,22 @@ if(MBEDIDE)
     set_property(GLOBAL PROPERTY MBED_STUDIO_ARM_COMPILER "--ide=mbed")
 endif()
 
+# Set the language standard to use per target
+function(mbed_set_language_standard target)
```

Comment:
The language standard is common. It should be defined once and used by all toolchains supported.

Maybe we should do the following instead:

1. In applications, ensure `add_subdirectory(${MBED_ROOT})` is one of the first things executed after setting paths variables (where we have it in Blinky is probably fine).
1. In mbed-os top level CMake input source file, set the default `CMAKE_TOOLCHAIN_FILE` as soon as we can (it needs to be before we define the `mbed-os` target).
1. In the `toolchain.cmake` module,  set the language for all targets using:
```
set(CMAKE_C_STANDARD 11)
set(CMAKE_C_STANDARD_REQUIRED YES)
set(CMAKE_C_EXTENSION YES)

set(CMAKE_CXX_STANDARD 14)
set(CMAKE_CXX_STANDARD_REQUIRED YES)
set(CMAKE_CXX_EXTENSION YES)
```

The key point is that the languages standard need to be set before any target is created so it applies to all of them.

Reply:
The point here was a function can't be in the `CMAKE_TOOLCHAIN_FILE ` as it's included by the system, not by our CMake lists (it should be available, I haven't yet figured out how to make it visible, will need to check CMake docs or the include of this toolchain file in cmake itself).

+1, I'll try to fix this.

Reply:
I am working on new PR , will address this one there.